libpermian/plugins/kickstart_test/__init__.py

```diff
@@ -184,6 +216,61 @@ def setup(self):
         else:
             LOGGER.info("Default rawhide installer boot.iso will be fetched.")
 
+    def process_scenario(self, scenario):
+        """Get test parameters from scenario event structure
+
+        :param scenario: structure holding scenario data
+        :type scenario: ScenarioStructure
+        :returns: tuple with
+                  - list of additional launcher arguments
+                  - path of tenmporary override defaults file to be used by launcher
+                  - boot.iso url based on installation tree location
+        :rtype: (list(str), str, str))
+        """
+        tree_boot_iso_url = None
+        command_args = []
+
+        installation_tree = None
+        platform = None
+        if scenario:
+            try:
+                installation_tree = scenario['installation_tree']
+            except KeyError:
+                pass
+            try:
+                platform = scenario['platform']
```

Comment:
I think platform should be taken from configuration in testplan (if you have separate testplans for different platforms). Alternatively it could be deduced from event itself, or different structure.